config/admin-functions/create-machine-cert.sh

```diff
@@ -0,0 +1,173 @@
+}
+
+function mount_usb() {
+    local usb_path="${1}"
+    unmount_usb_if_mounted
+    "${VX_METADATA_ROOT}/app-scripts/mount-usb.sh" "${usb_path}"
+}
+
+function mount_usb_if_present() {
+    local usb_path="$(get_usb_path)"
+    if [[ -n "${usb_path}" ]]; then
+        mount_usb "${usb_path}"
+    fi
+}
+
+function wait_for_usb_and_mount_once_present() {
+    while true; do
+        local usb_path="$(get_usb_path)"
+        if [[ -n "${usb_path}" ]]; then
+            mount_usb "${usb_path}"
+            break
+        fi
+        sleep 1
+    done
+}
+
+# OpenSSL helpers
+
+function generate_private_key() {
+    local private_key_path="${1}"
+    local private_key_password="${2}"
+    openssl ecparam -genkey -name prime256v1 -noout | \
+        openssl pkcs8 -topk8 -passout "pass:${private_key_password}" -out "${private_key_path}"
+}
+
+function construct_machine_cert_subject() {
+  local machine_jurisdiction="${1:-}"
+  if [[ -n "${machine_jurisdiction}" ]]; then
+    echo "/C=US/ST=CA/O=VotingWorks/${VX_IANA_ENTERPRISE_OID}.1=${VX_MACHINE_TYPE}/${VX_IANA_ENTERPRISE_OID}.2=${machine_jurisdiction}/"
+  else
+    echo "/C=US/ST=CA/O=VotingWorks/${VX_IANA_ENTERPRISE_OID}.1=${VX_MACHINE_TYPE}/"
+  fi
+}
+
+function create_machine_cert_signing_request() {
+    local machine_private_key_path="${1}"
+    local machine_private_key_password="${2}"
+    local machine_cert_subject="${3}"
+    local cert_signing_request_path="${4}"
+    openssl req -new \
+        -config "${VX_METADATA_ROOT}/vxsuite/libs/auth/certs/openssl.cnf" \
+        -key "${MACHINE_PRIVATE_KEY_PATH}" \
+        -passin "pass:${machine_private_key_password}" \
+        -subj "${machine_cert_subject}" \
+        -out "${cert_signing_request_path}"
+}
+
+# ---------- Script ----------
+
+# TODO: Use the machine's TPM private key (generated by the generate-key.sh script) instead of a
+# new key
+machine_private_key_password="$(get_machine_private_key_password_from_user_input)"
+generate_private_key "${MACHINE_PRIVATE_KEY_PATH}" "${machine_private_key_password}"
+
+if [[ ${VX_MACHINE_TYPE} == "admin" ]]; then
+    machine_jurisdiction="$(get_machine_jurisdiction_from_user_input)"
+    machine_cert_subject="$(construct_machine_cert_subject "${machine_jurisdiction}")"
+else
+    machine_cert_subject="$(construct_machine_cert_subject)"
+fi
+
+echo "Insert a USB into the machine. The USB will be auto-detected."
+sleep 1
+wait_for_usb_and_mount_once_present
+echo "USB detected!"
+
+echo "Writing cert signing request to USB..."
+rm -rf "${USB_CERTS_DIRECTORY}"
+mkdir "${USB_CERTS_DIRECTORY}"
+create_machine_cert_signing_request \
+    "${MACHINE_PRIVATE_KEY_PATH}" \
+    "${machine_private_key_password}" \
+    "${machine_cert_subject}" \
+    "${USB_CERTS_DIRECTORY}/csr.pem"
+unmount_usb
+
+read -p "Remove the USB, take it to VxCertifier, and bring it back to this machine when prompted. Press enter once you've re-inserted the USB. "
```

Comment:
Will be adding/documenting the openssl command run by VxCertifier in a separate PR